src/db.c

```diff
@@ -307,10 +307,12 @@ robj *dbRandomKey(redisDb *db) {
 
 /* Helper for sync and async delete. */
 static int dbGenericDelete(redisDb *db, robj *key, int async) {
```

Comment:
what about dbOverwrite?
should this event be fired in the same places `moduleNotifyKeyUnlink` is called?

Reply:
Maybe something in `dbOverwrite` is wrong. e.g. in `spopWithCountCommand`, this is really not unlink + add.

Reply:
yes
dbOverwrite in spopWithCountCommand (and maybe others) should not fire the event
dbOverwite in setKey (and maybe others) should fire the event

Reply:
yes, as i commented here https://github.com/redis/redis/pull/9406#discussion_r979359382
i think we should keep the same approach we had when this was a KSN, and call it from moduleNotifyKeyUnlink.
the only difference is that we took the hook out of KSN, but it's still something that needs to be done every time a key is removed from the database.
the only thing extra i think it would be nice to add a distinction between DELETE, OVERWRITE, EXPIRED, and EVICTED.

Reply:
@oranagra Can you confirm whether I get your point with the lastest commit (without tests and expired evicted).
If do, I will continue.

Reply:
yes, i think that's what i meant.
it's actually like your original PR, just shifting from KSN to the event infra.
p.s. you should pass `key` to the event struct, not `key-ptr`.